apps/protailwind/src/pages/invoices/[merchantChargeId].tsx

```diff
@@ -43,6 +43,18 @@ export const getServerSideProps: GetServerSideProps = async ({
       )
       const bulkCoupon = purchase && purchase.bulkCoupon
 
+      let quantity = 1
+      if (purchase?.merchantPurchase?.checkoutSessionId) {
+        const checkoutSession = await stripe.checkout.sessions.retrieve(
+          purchase?.merchantPurchase?.checkoutSessionId,
+          {expand: ['line_items']},
+        )
+
+        quantity = checkoutSession.line_items?.data[0].quantity || 1
+      } else if (bulkCoupon) {
+        quantity = bulkCoupon.maxUses
+      }
```

Comment:
determine (server-side) the `quantity` to be used in the invoice (rendered below)

We attempt to look up the quantity from Stripe via the checkout session ID. If that is unavailable, then we fallback to the `bulkCoupon.maxUses` which is going to work for most existing purchases anyway.